### PR TITLE
72 - swipe to refresh in categories fragment

### PR DESCRIPTION
**Pull request by VVMaxVV**
Added swipe to refresh layout on Category and Product List Fragment
https://trello.com/c/7DbiHRFy/72-swipe-to-refresh-in-categories-fragment

### CODE REVIEW

File: app/src/main/java/com/example/internetshop/presentation/viewModel/CategoriesViewModel.kt [hunk 1]
```diff
@@ -19,16 +19,19 @@ class CategoriesViewModel @Inject constructor(
 
     val navEventLiveData = SingleLiveEvent<CategoryEvent>()
     val categoriesLiveData = MutableLiveData<List<CategoryViewState>>()
+    val progressBarLiveData = MutableLiveData<Boolean>()
```
Comment (darthoo) [inline app/src/main/java/com/example/internetshop/presentation/viewModel/CategoriesViewModel.kt:22]: Try to not use `liveData` in variable names. It's just redundant. `progress` is enough information

File: app/src/main/java/com/example/internetshop/presentation/viewModel/ProductsListViewModel.kt [hunk 2]
```diff
@@ -17,12 +17,17 @@ class ProductsListViewModel @Inject constructor(
 ) : BaseViewModel() {
     val productsList = MutableLiveData<List<ProductViewState>>()
     val navEventLiveData = SingleLiveEvent<Event>()
+    val progressBarLiveData = MutableLiveData<Boolean>()
```
Comment (darthoo) [inline app/src/main/java/com/example/internetshop/presentation/viewModel/ProductsListViewModel.kt:20]: Same as above

File: app/src/main/res/layout/item_product_category.xml [hunk 3]
```diff
@@ -22,7 +22,8 @@
             android:layout_height="wrap_content"
             android:layout_gravity="center_horizontal"
             android:background="@color/white"
-            android:onClick="@{()->product.onClick(product.id, product.title)}">
+            android:onClick="@{()->product.onClick(product.id, product.title)}"
+            android:padding="8dp">
```
Comment (darthoo) [inline app/src/main/res/layout/item_product_category.xml:26]: dimens

File: app/src/main/res/layout/item_product_category.xml [hunk 4]
```diff
@@ -67,7 +68,7 @@
                 android:layout_width="wrap_content"
                 android:layout_height="wrap_content"
                 android:layout_marginTop="@dimen/dimen_item_product_category_price_top_margin"
-                android:text="@{product.price + '$'}"
+                android:text="@{product.price + @string/currency}"
```
Comment (darthoo) [inline app/src/main/res/layout/item_product_category.xml:71]: Please use parametrized string 
Here's examples:
https://developer.android.com/guide/topics/resources/string-resource#formatting-strings
https://stackoverflow.com/questions/38978499/how-do-i-use-databinding-to-combine-a-string-from-resources-with-a-dynamic-varia

**Review by darthoo — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/internetshop/presentation/viewModel/CategoriesViewModel.kt (hunk 1 above)
Comment (VVMaxVV) [reply app/src/main/java/com/example/internetshop/presentation/viewModel/CategoriesViewModel.kt:22]: fixed

File: app/src/main/java/com/example/internetshop/presentation/viewModel/ProductsListViewModel.kt (hunk 2 above)
Comment (VVMaxVV) [reply app/src/main/java/com/example/internetshop/presentation/viewModel/ProductsListViewModel.kt:20]: fixed

File: app/src/main/res/layout/item_product_category.xml (hunk 3 above)
Comment (VVMaxVV) [reply app/src/main/res/layout/item_product_category.xml:26]: fixed

File: app/src/main/res/layout/item_product_category.xml (hunk 4 above)
Comment (VVMaxVV) [reply app/src/main/res/layout/item_product_category.xml]: fixed

**Pull request merged**